Add wrapper for get record API call

cecilia-donnelly opened the pull request:
Also add an example script to show how to get the file url, aka the
CDN link.

Thanks to @meisekimiu for writing the copy-publish example script, which I used to structure this example, too.

cecilia-donnelly commented:
(Sorry to add more functionality before we do our design work, but this is pretty minimal and should be helpful for HyperAudio.)

meisekimiu commented on examples/get-cdn-link.js:
```diff
@@ -0,0 +1,46 @@
+    const randomChild = randomElement(privateRoot.ChildItemVOs);
+
+    // doing this specifically for a record
+    if (randomChild && randomChild.uploadFileName) {
```
To some degree I realize that this design is hopefully going to be completely redone anyways, so we shouldn't be too nitpicky about any of these smaller changes, but can we have a publicly exposed helper function that's something like `permanent.record.isRecord` which specifically tests for the existence of `uploadFileName`? I don't think the user should have to do that here.

To avoid the issue that we ran into earlier this morning, I'd also suggest initially filtering out all the non-folder items to begin with so we can have a more useful error message and a folder isn't randomly selected forcing us to restart the script, maybe something like this:
```javascript
const childRecords = privateRoot.ChildItemVOs.filter((item) => permanent.record.isRecord(item)); // or whatever an isRecord helper is called
if (childRecords.length === 0) {
  console.log("No records in My Files folder");
} else {
  const randomChild = randomElement(childRecords);
  // ... insert everything else here ...
}

console.log('Done!');
```

cecilia-donnelly replied:
Given that you've already written `isItemARecord` I'm inclined to use it again.  I see it's protected - is that for a bigger reason that I don't know or because when you initially used it you only wanted it in the item resource?

cecilia-donnelly replied:
(Also, sorry, I skipped over the "yes this is a much better/more useful way to present this example!"  And thank you!)

meisekimiu replied:
I believe IsItemARecord is only protected because I was only using it internally. Feel free to make it public!

meisekimiu commented on src/lib/resources/record.resource.ts:
```diff
@@ -129,4 +129,14 @@ export class RecordResource extends BaseResource {
     return this.getVoFromResponse<RecordResponse>(response, 'RecordVO');
   }
+
+  public async getRecordById(recordId: number): Promise<RecordVO> {
```
Do we want test coverage for this? I actually don't know if we do or not.

cecilia-donnelly replied:
I think it's a good idea!

cecilia-donnelly commented:
Okay @meisekimiu I believe I've addressed your comments.  Let me know how it looks.  This is my first time writing a test in the node-sdk (I think?) so I based it on the tests I saw for earlier `record` methods.  Is it useful in this form?

Thanks again for your review and your patience!

meisekimiu reviewed (approved):
Looks good!
